R/plot.R

```diff
@@ -106,10 +108,10 @@ plot.sf <- function(x, y, ..., ncol = 10, col = NULL, max.plot = 9, main) {
 		}
 		# col selection may have changed; set cols again:
 		cols = setdiff(names(x), attr(x, "sf_column"))
-		invisible(lapply(cols, function(cname) plot(x[, cname], main = cname, col = col, ...)))
+		invisible(lapply(cols, function(cname) plot(x[, cname], main = cname, col = col, pal = pal, ...)))
 	} else {
 		if (is.null(col) && ncol(x) == 2)
-			col = sf.colors(ncol, x[[setdiff(names(x), attr(x, "sf_column"))]])
+			col = get_colors(pal, ncol, x[[setdiff(names(x), attr(x, "sf_column"))]])
```

Comment:
Why not use `pal(ncol, x[[setdiff(names(x), attr(x, "sf_column"))]])` here?

Reply:
Not all palettes take two arguments in this way. For example, `heat.colors` does not.